compiler/rustc_typeck/src/errors.rs

```diff
@@ -324,3 +324,20 @@ pub struct SubstsOnOverriddenImpl {
     #[primary_span]
     pub span: Span,
 }
+
+pub struct RustcOutlives {
+    pub span: Span,
+    pub pred: Vec<String>,
+}
+
+impl SessionDiagnostic<'_> for RustcOutlives {
```

Comment:
This should not use a manual implementation of `SessionDiagnostic` but instead be using the derive macro, I think.

Reply:
Yeah, you will be able to simplify this with a diagnostic derive, you can see examples [in the blog post](https://blog.rust-lang.org/inside-rust/2022/08/16/diagnostic-effort.html#using-a-diagnostic-derive).